src/main/groovy/io/spring/gradle/convention/SpringSampleWarPlugin.groovy

```diff
@@ -73,8 +76,8 @@ public class SpringSampleWarPlugin extends SpringSamplePlugin {
 			def gretty = project.gretty
 			String host = project.gretty.host ?: 'localhost'
 			boolean isHttps = gretty.httpsEnabled
-			int httpPort = gretty.httpPort
-			int httpsPort = gretty.httpsPort
+			Integer httpPort = integrationTest.systemProperties['gretty.httpPort']
```

Comment:
Why did this need to be changed?

Reply:
The builds are failing due to a port collision when running gretty-based samples. Or at least, the last two failures have been this.

While unlikely, since the code is configuring several random ports at configuration time, it's possible that one of them will get taken up by something else before those port "reservations" are actually exercised. This race condition is eliminated if we allow ServerSocket to pick the port when the actual server is being started up.

Reply:
That makes sense. I was more specifically trying to ask...Why are we getting it from the system properties vs from gretty though?

Reply:
This specific line needed changing because gretty does not update the original configuration object once the server is started. Instead, it sets a system property with the selected port.